### PR TITLE
Added iOS Task Options, Task Result and Error Codes

### PR DESCRIPTION
**Pull request by priankakariatyml**

### CODE REVIEW

File: mediapipe/tasks/ios/core/sources/MPPTaskOptions.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * @discussion The external model file must be a single standalone TFLite file. It could be packed
+ * with TFLite Model Metadata[1] and associated files if they exist. Failure to provide the
+ * necessary metadata and associated files might result in errors. Check the [documentation]
+ * (https://www.tensorflow.org/lite/convert/metadata) for each task about the specific requirement.
+ *
+ * @param modelPath An absolute path to a TensorFlow Lite model file stored locally on the device.
+ *
+ * @return An instance of `MPPTaskOptions` initialized to the given model path.
+ */
+- (instancetype)initWithModelPath:(NSString *)modelPath;
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/core/sources/MPPTaskOptions.h:44]: This API currently does not exist in the C++ or Android API. Do you think we need it for iOS?

File: mediapipe/tasks/ios/core/sources/MPPTaskResult.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * MediaPipe Tasks options base class. Any MediaPipe task-specific options class should extend
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/core/sources/MPPTaskResult.h:20]: This comment mentions "task options" but the class is "task result".

File: mediapipe/tasks/ios/core/sources/MPPTaskResult.h [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * this class.
+ */
+NS_SWIFT_NAME(TaskResult)
+@interface MPPTaskResult : NSObject <NSCopying>
```
Comment (schmidt-sebastian) [inline mediapipe/tasks/ios/core/sources/MPPTaskResult.h:24]: This class seems to exist only in iOS. Is this needed or can we follow the other implementations more closely?

**Review by schmidt-sebastian — CHANGES_REQUESTED** [review]
Thank you for sending this. Some comments.

File: mediapipe/tasks/ios/core/sources/MPPTaskOptions.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * Base options for configuring the Mediapipe task.
+ */
+@property(nonatomic, copy) MPPBaseOptions *baseOptions;
+
+/**
+ * Initializes a new `MPPTaskOptions` with the absolute path to the model file
+ * stored locally on the device, set to the given the model path.
+ *
+ * @discussion The external model file must be a single standalone TFLite file. It could be packed
```
[the remark this answers is not in the document]
Comment (priankakariatyml) [reply mediapipe/tasks/ios/core/sources/MPPTaskOptions.h]: I have rewritten this.

File: mediapipe/tasks/ios/core/sources/MPPTaskResult.h (hunk 3 above)
Comment (priankakariatyml) [reply mediapipe/tasks/ios/core/sources/MPPTaskResult.h:25]: Following the Java implementation here.

File: mediapipe/tasks/ios/core/sources/MPPTaskOptions.h (hunk 1 above)
Comment (priankakariatyml) [reply mediapipe/tasks/ios/core/sources/MPPTaskOptions.h]: It's just an additional convenience initialiser for initialising the class.  I have removed it for now. We can add back if need be

**Review by schmidt-sebastian — APPROVED** [review]

**Pull request merged**